test/uk/gov/hmrc/tai/service/UpdateNextYearsIncomeServiceSpec.scala

```diff
@@ -0,0 +1,206 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.service
+
+import org.mockito.Matchers
+import org.mockito.Mockito.{times, verify, when}
+import org.mockito.Matchers.any
+import org.scalatest.mock.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import uk.gov.hmrc.domain.{Generator, Nino}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.model.domain.income._
+import uk.gov.hmrc.tai.model.domain.{Employment, EmploymentIncome}
+import uk.gov.hmrc.tai.util.constants.journeyCache.UpdateNextYearsIncomeConstants
+import uk.gov.hmrc.time.TaxYearResolver
+import scala.concurrent.duration._
+
+import scala.concurrent.{Await, Future}
+import scala.util.Random
+
+class UpdateNextYearsIncomeServiceSpec extends PlaySpec with MockitoSugar {
+
+  "setup" must {
+    "initialize the journey cache and return the cache model" when {
+      "an taxCodeIncome and Employment is returned" in {
+        val nino = generateNino
+
+        when(updateNextYearsIncomeService.employmentService.employment(Matchers.eq(nino), Matchers.eq(employmentId))(any()))
+          .thenReturn(Future.successful(Some(employment(employmentName))))
+
+        when(updateNextYearsIncomeService.taxAccountService.taxCodeIncomeForEmployment(
+          Matchers.eq(nino), Matchers.eq(TaxYear()), Matchers.eq(employmentId))(any())
+        ).thenReturn(Future.successful(Some(taxCodeIncome(employmentName, employmentId, employmentAmount))))
+
+        val result = Await.result(updateNextYearsIncomeService.setup(employmentId, nino), 5.seconds)
+
+        verify(updateNextYearsIncomeService.journeyCacheService, times(1))
+          .cache(expectedMap(employmentName, employmentId, employmentAmount))
+
+        result mustBe UpdateNextYearsIncomeCacheModel(employmentName, employmentId, employmentAmount, None)
+      }
+    }
+
+    "throw a runtime exception" when {
+      "could not retrieve a TaxCodeIncome" in {
+        val nino = generateNino
+
+        when(updateNextYearsIncomeService.employmentService.employment(Matchers.eq(nino), Matchers.eq(employmentId))(any()))
+          .thenReturn(Future.successful(Some(employment(employmentName))))
+
+        when(updateNextYearsIncomeService.taxAccountService.taxCodeIncomeForEmployment(
+          Matchers.eq(nino), Matchers.eq(TaxYear()), Matchers.eq(employmentId))(any())
+        ).thenReturn(Future.successful(None))
+
+        val result: RuntimeException = the[RuntimeException] thrownBy Await.result(updateNextYearsIncomeService.setup(employmentId, nino), 5 seconds)
```

Comment:
I _think_ we need to intercept this still or else it pollutes the output of the tests

Reply:
Can you elaborate on this

Reply:
Im not seeing any bad output from the tests
```
[info] Configuring Artifactory... Host: not set. User: not set. Password defined: false
[info] Compiling 2 Scala sources to /Users/seanarmstrong/Development/tai-frontend/target/scala-2.11/classes...
[info] UpdateNextYearsIncomeServiceSpec:
[info] setup
[info]   must initialize the journey cache and return the cache model
[info]   - when an taxCodeIncome and Employment is returned
[info]   must throw a runtime exception
[info]   - when could not retrieve a TaxCodeIncome
[info]   - when could not retrieve a Employment
[info] get
[info]   must retrieve the cached values and returns the cache model
[info]   - when journey values exist without a new amount in the cache
[info]   - when journey values exist with a new amount in the cache
[info]   must setup the cache
[info]   - when journey values do not exist in the cache
[info] setNewAmount
[info]   must update the cache with the new amount
[info]   - when journey values are successfully retrieved from the cache
[info] ScalaCheck
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] Warning: Unknown ScalaCheck args provided: -o -u target/test-reports -h target/test-reports/html-report -o -u target/test-reports -h target/test-reports/html-report
[info] ScalaTest
[info] Run completed in 3 seconds, 377 milliseconds.
[info] Total number of tests run: 7
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 7, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```